server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java

```diff
@@ -301,7 +301,8 @@ public void testSimpleTerminateAfterCount() throws Exception {
         SearchResponse searchResponse;
         int size;
         for (int i = 1; i < max; i++) {
-            size = randomIntBetween(0, max);
+            // Size=0 has known issues, see https://github.com/opensearch-project/OpenSearch/issues/10435
+            size = randomIntBetween(1, max);
```

Comment:
nit: how about we replicate the test logic in a separate method which takes in `size` as input parameter. Then have 2 tests which calls this method with different size argument. One with `randomIntBetween(1, max)` and other with size=0. Then we can just add `AwaitsFix` annotation  for the second test

Reply:
This makes sense to me, especially since `size=0` is clearly a unique case we should have coverage for it all the time instead of randomly. Will make this change in the next revision.